docs/img/factory_dispatching.svg

```diff
@@ -1,3 +1,3 @@
 height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 38px; height: 1px; padding-top: 265px; margin-left: 1316px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 45px"><b>.</b></font></div></div></div></foreignObject><text x="1335" y="269" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">.</text></switch></g><rect x="1315" y="275" width="40" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 38px; height: 1px; padding-top: 285px; margin-left: 1316px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 45px"><b>.</b></font></div></div></div></foreignObject><text x="1335" y="289" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">.</text></switch></g><rect x="1620" y="0" width="310" height="60" rx="9" ry="9" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 308px; height: 1px; padding-top: 30px; margin-left: 1621px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 30px" color="#ffffff">PandasOnRayIO</font></div></div></div></foreignObject><text x="1775" y="34" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">PandasOnRayIO</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```

Comment:
It appears that the "OmnisciOnNativeFactory" name is too long for the textbox and so the string comes out of its borders. I've expanded borders in the suggested changes. 

P.S. since suggested changes modifies SVG code itself, please verify the resulted image.

![image](https://user-images.githubusercontent.com/62142979/133751065-16e88c2a-9e8d-486c-9be8-381c2896e157.png)

```suggestion
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="1931px" height="305px" viewBox="-0.5 -0.5 1931 305" content="&lt;mxfile host=&quot;app.diagrams.net&quot; modified=&quot;2021-09-17T08:26:32.709Z&quot; agent=&quot;5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.82 Safari/537.36&quot; etag=&quot;cFIBiBuFmrKV9RD3L8Zm&quot; version=&quot;14.6.10&quot; type=&quot;google&quot;&gt;&lt;diagram id=&quot;Oc6d4yEnh7qIssPu6n0P&quot; name=&quot;Page-1&quot;&gt;7Vpdd5s4EP01fkwOIMD2Y+I46Z7dNtlNz9n2qUcF2WhXICqEbfrrK0DiOzaJje008YvRaCTw3Jk7V7ZHYOZv7hgMvY/URWRkaO5mBG5GhmFMJrp4Sy1JbtGBZueWJcOutJWGR/wTSaMmrTF2UVRz5JQSjsO60aFBgBxes0HG6LrutqCkftcQLlHL8OhA0rb+i13u5daJpZX2DwgvPXVnXZMzPlTO0hB50KXrignMR2DGKOX5lb+ZIZJGT8UlX3f7xGzxYAwFvM+CT3/C+Wd/nVA+5XP+4WHxLVhc6JZ8OJ6oT4xcEQA5pIx7dEkDSOal9dqJ2Qqlu+piULr8RWkojf8hzhMJJow5FSaP+0TOog3mXyrXX8W1dmlYcniTZo+mBokaBJwlX6qD6rJ0XK7LRmphxBn9vwAv9W5HTgYzojFz0LZwyQyEbIn4Fj8j90tDWbmBxOUOUR+JJxQODBHI8aqea1Cm7LLwK1EVFxLY54Cc77uCJJZ3Ghk2Ec9/vaAiAlX07R8xVRMXUQbglXAAdrgpJ8XVMn0X5Y4DMRtHiKkdxQPmm+YurexiNA7cInnWHuboMYRZ0NeCQuqJku40o4SybC1YZK/Ujgmp2DXtWrvVtgG7QoyjzVYo5OyFPpH1qhhLjddl+Rfs5FVLX1X+4fGzT1ikWq1Ixz2LVK8X6fiYRWr0LFL9vIp08saY2DwKyOCsQDYGYuLQvWQIut+caHVKIjbtsTHbimt/ItanZ8fDhZJ75+E+JQpepVgC47Ph4ZfR8DFZuC/E4OA0nC29YgwmFYeQ4oBHlZ0fUkPJKVMFruQUIR3rR6Yd/sBqpFf+AGWyFZ9kj/w7eIsQZ2PF0KVRcXirldxCh1ORLjgKIXe8oZT9cI3D1nY3DnBkAT/eG1Sjs++rXb4rww3k8JZBH1VQ+9703oWkCDSvw5VzhgIsoAFqYChNkOBlIIaOgE9kDrhOYcMOJFdywseumzFmV36UGXSww1w9FYwuDWF1pIIxmISYdmRCs90QgsPoqSA9iUp2EE5fQx+QjUZ96e2gTjtiOh0qpkBvh/B4sqzo0l8rM90d++V9VhHI7jPt+LzE1P4Hnp7E93eMWDKjfohJrWW9XfIDk3MjPzB9U9J6z9PTMJK5pXF106zniRS5xRY558hVjQw4hN49gODV9hG8DzBwYXQf/AMTqX1fm+LVtYbktdqC1+4odHuwQjd3YdoToedjn0N17wc4cvB98Ckrjd8F12nHSea4wFonBnYWu4vfqlQ7T6dHxdTc/+dB0+ql0S7fdVk6q9sNYQbaKWAeU5eZO+n6PQOGzQDr1Bmwk9ffM2DYDBifOAOsyfnI8D/uX11bbx6iOhR411fOL2jqYlj+Zyw/hJV/vQPzXw==&lt;/diagram&gt;&lt;/mxfile&gt;" style="background-color: rgb(255, 255, 255);"><defs/><g><path d="M 210 200 Q 210 200 356.16 200" fill="none" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 364.41 200 L 353.41 205.5 L 356.16 200 L 353.41 194.5 Z" fill="#000000" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="all"/><rect x="0" y="175" width="210" height="100" rx="15" ry="15" fill="#00b0f0" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 208px; height: 1px; padding-top: 225px; margin-left: 1px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #ffffff; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 36px">Modin user</font></div></div></div></foreignObject><text x="105" y="229" fill="#ffffff" font-family="Helvetica" font-size="12px" text-anchor="middle">Modin user</text></switch></g><path d="M 370 250 Q 370 250 223.84 250" fill="none" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 215.59 250 L 226.59 244.5 L 223.84 250 L 226.59 255.5 Z" fill="#000000" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="all"/><path d="M 580 200 Q 580 200 768.03 200" fill="none" stroke="#000000" stroke-width="4" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 775.53 200 L 765.53 205 L 768.03 200 L 765.53 195 Z" fill="#000000" stroke="#000000" stroke-width="4" stroke-miterlimit="10" pointer-events="all"/><rect x="370" y="175" width="210" height="100" rx="15" ry="15" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 208px; height: 1px; padding-top: 225px; margin-left: 371px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #ffffff; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 36px">pd.read_csv</font></div></div></div></foreignObject><text x="475" y="229" fill="#ffffff" font-family="Helvetica" font-size="12px" text-anchor="middle">pd.read_csv</text></switch></g><path d="M 780 250 Q 780 250 593.84 250" fill="none" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 585.59 250 L 596.59 244.5 L 593.84 250 L 596.59 255.5 Z" fill="#000000" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="all"/><path d="M 1090 225 Q 1117 225 1117 127.5 Q 1117 30 1168.03 30" fill="none" stroke="#000000" stroke-width="4" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 1175.53 30 L 1165.53 35 L 1168.03 30 L 1165.53 25 Z" fill="#000000" stroke="#000000" stroke-width="4" stroke-miterlimit="10" pointer-events="all"/><rect x="780" y="175" width="310" height="100" rx="15" ry="15" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 308px; height: 1px; padding-top: 225px; margin-left: 781px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 36px" color="#ffffff">FactoryDispatcher</font></div></div></div></foreignObject><text x="935" y="229" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">FactoryDispatcher</text></switch></g><rect x="170" y="275" width="250" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 248px; height: 1px; padding-top: 285px; margin-left: 171px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 26px"><b>DataFrame</b></font></div></div></div></foreignObject><text x="295" y="289" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">DataFrame</text></switch></g><ellipse cx="105" cy="150" rx="45" ry="45" fill="#00b0f0" stroke="#000000" pointer-events="all"/><rect x="560" y="275" width="250" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 248px; height: 1px; padding-top: 285px; margin-left: 561px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 26px"><b>QueryCompiler</b></font></div></div></div></foreignObject><text x="685" y="289" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">QueryCompiler</text></switch></g><path d="M 1540 30 Q 1540 30 1606.16 30" fill="none" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 1614.41 30 L 1603.41 35.5 L 1606.16 30 L 1603.41 24.5 Z" fill="#000000" stroke="#000000" stroke-width="5" stroke-miterlimit="10" pointer-events="all"/><rect x="1180" y="0" width="360" height="60" rx="9" ry="9" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 358px; height: 1px; padding-top: 30px; margin-left: 1181px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 30px" color="#ffffff">PandasOnRayFactory</font></div></div></div></foreignObject><text x="1360" y="34" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">PandasOnRayFactory</text></switch></g><rect x="1180" y="85" width="360" height="60" rx="9" ry="9" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 358px; height: 1px; padding-top: 115px; margin-left: 1181px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font color="#ffffff" style="font-size: 30px">OmnisciOnNativeFactory</font></div></div></div></foreignObject><text x="1360" y="119" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">OmnisciOnNativeFactory</text></switch></g><rect x="1180" y="175" width="360" height="60" rx="9" ry="9" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 358px; height: 1px; padding-top: 205px; margin-left: 1181px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font color="#ffffff" style="font-size: 30px">CudfOnRayFactory</font></div></div></div></foreignObject><text x="1360" y="209" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">CudfOnRayFactory</text></switch></g><rect x="1340" y="225" width="40" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 38px; height: 1px; padding-top: 235px; margin-left: 1341px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 45px"><b>.</b></font></div></div></div></foreignObject><text x="1360" y="239" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">.</text></switch></g><rect x="1340" y="245" width="40" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 38px; height: 1px; padding-top: 255px; margin-left: 1341px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 45px"><b>.</b></font></div></div></div></foreignObject><text x="1360" y="259" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">.</text></switch></g><rect x="1340" y="265" width="40" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 38px; height: 1px; padding-top: 275px; margin-left: 1341px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 45px"><b>.</b></font></div></div></div></foreignObject><text x="1360" y="279" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">.</text></switch></g><rect x="1620" y="0" width="310" height="60" rx="9" ry="9" fill="#4672c4" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 308px; height: 1px; padding-top: 30px; margin-left: 1621px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font style="font-size: 30px" color="#ffffff">PandasOnRayIO</font></div></div></div></foreignObject><text x="1775" y="34" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">PandasOnRayIO</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```

Reply:
I checked - everything is fine, thanks!